akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowDedupeSpec.scala

```diff
@@ -0,0 +1,32 @@
+/**
+  * Copyright (C) 2016 Typesafe Inc. <http://www.typesafe.com>
+  */
+package akka.stream.scaladsl
+
+import akka.stream.testkit._
+import akka.stream.testkit.Utils.assertAllStagesStopped
+import akka.stream.testkit.scaladsl.{ TestSource, TestSink }
+import akka.stream.{ ActorMaterializer, ActorMaterializerSettings }
+
+class FlowDedupeSpec extends AkkaSpec {
+
+  implicit val materializer = ActorMaterializer()
+
+  "A Dedupe" must {
+
+    "remove consecutive duplicates" in {
+      val input = List(1, 1, 1, 2, 2, 1, 1, 3)
+      val probe = TestSubscriber.manualProbe[Int]()
+      Source(input).dedupe.runWith(Sink.fromSubscriber(probe))
+
+      val subscription = probe.expectSubscription()
+      subscription.request(Long.MaxValue)
+
+      probe.expectNext(1)
```

Comment:
On further analysis I think it ought to be named `destutter`:

https://wpollock.com/lister.php?file=AJava/Destutter.java&nodir&linenums&dl
https://books.google.com/books?id=nabtAQAAQBAJ&pg=PA63&lpg=PA63&dq=%22destutter%22&source=bl&ots=yZp5XRI5lZ&sig=L7p4F1mxiJ6ot52MpOzhezo9K28&hl=en&sa=X&ved=0ahUKEwim1P-JoqTKAhWJtxoKHU-TAvEQ6AEILTAD#v=onepage&q=%22destutter%22&f=false
https://gist.github.com/invisiblefunnel/1248809

and more